tensorrt_llm/llmapi/llm_args.py

```
@@ -457,6 +458,42 @@ def num_capture_layers(self):
+        return cls(**data)
+
+    decoding_type: ClassVar[str] = "SaveState"
+
+    def validate(self) -> None:
+        if self.output_directory is None or not self.eagle3_layers_to_capture:
+            raise ValueError(
+                "Save directory and layers to capture must be provided")
+
+    @functools.cached_property
+    def spec_dec_mode(self):
+        from tensorrt_llm._torch.speculative.interface import \
```

Ruff (0.12.2)

1 blank line required between summary line and description

(D205)